public/jwpltx.js

```diff
@@ -98,27 +165,51 @@ window.jwpltx = window.jwpltx || {};
         } else {
           uri.q = 32;
         }
-        uri.ti = Math.round(uri.vd / uri.q);
+
+        uri.ti = 0;
         uri.pw = 0;
-        sendData('s');
 
+        // Initialize latest quantile to compare further quantiles with
+        nextQuantile = getNextTriggerQuantile(vp, vd);
+        // Initial values to compare watched progress
+        lastVp = vp;
+
+        sendData('s');
         // monitor ticks for entering new quantile
       } else {
-        let pw = (Math.floor(vp / uri.ti) * 128) / uri.q;
-        if (pw != uri.pw) {
+        const pw = getCurrentProgressQuantile(vp, vd);
+        const quantile = getNextTriggerQuantile(vp, vd);
+
+        // Total time watched since last t event.
+        timeWatched = timeWatched + (vp - lastVp);
+        lastVp = vp;
+
+        if (pw >= nextQuantile) {
+          uri.ti = Math.round(timeWatched);
           uri.pw = pw;
+
           sendData('t');
+
+          nextQuantile = quantile;
+          timeWatched = 0;
         }
       }
     }
   };
 
   // Process a video complete events
   o.complete = function () {
-    if (uri.pw != 128) {
-      uri.pw = 128;
-      sendData('t');
+    // Clear intervals for live streams
+    if (uri.pw === -1) {
+      clearLiveInterval();
+    } else {
+      uri.pw = MAX_DURATION_IN_QUANTILES;
     }
+
```

Comment:
Should we also clear the seek timeout here just in case they seeked to the end or almost the end?

Reply:
Does it potentially need to be cleared here too?

Reply:
Missed this place, thought that `sendRemainingData` had to cover it.